token.sol

```diff
@@ -237,13 +235,14 @@ contract token is safeMath, module, announcementTypes {
         if ( _from != msg.sender ) {
             allowance_[_from][msg.sender].amount = safeSub(allowance_[_from][msg.sender].amount, _amount);
             allowance_[_from][msg.sender].transactionCount++;
-            EAllowanceUsed(msg.sender, _from, _amount);
+            AllowanceUsed(msg.sender, _from, _amount);
         }
         if ( isContract(_to) ) {
             bytes memory data;
             transferToContract(_from, _to, _amount, data);
         } else {
-            require( transfer_( _from, _to, _amount, true) );
+            transfer_( _from, _to, _amount, true);
+            Transfer(_from, _to, _amount);
```

Comment:
Empty `_data` should be logged into blockchain if you are using ERC223 `Transfer` event.
In all cases, you should use similar events. If you are using Transfer event with _data then you should use this events in all transfers and transferFroms.
You can fire two events (one with data and the another without it) but not one type of the event for transfer and the another type of event for transferFrom.